src/twisted/internet/test/test_tcp.py

```diff
@@ -750,6 +765,189 @@ def whenRun():
             raise SkipTest("Reactor does not support ITLSTransport")
         self.assertEqual(BrokenContextFactory.message, str(results[0]))
 
+    def test_connectPortUnderflow(self):
```

Comment:
New code / new tests should have type annotations.
```suggestion
    def test_connectPortUnderflow(self) -> None:
```

Reply:
(If it would be a big burden to annotate this stuff because it's surfacing whole new classes of typing problem, I would say it's fine to just file an issue to note the issue and not block here, particularly given that these are tests rather than new API)